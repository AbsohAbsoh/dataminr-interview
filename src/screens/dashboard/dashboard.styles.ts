import {StyleSheet} from 'react-native';
import {Theme} from '../../config/theme';

export const dashboardStyles = (theme: Theme) => {
  return StyleSheet.create({
    background: {
      backgroundColor: theme.background,
      flex: 1,
      padding: 25,
      width: '100%',
      alignItems: 'center',
    },
    demoParagraph: {
      marginTop: 150,
      textAlign: 'center',
    },
    demoButton: {
      marginTop: 150,
    },
  });
};
