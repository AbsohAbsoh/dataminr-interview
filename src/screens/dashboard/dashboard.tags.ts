export const dashboardTags = {
  demoButton: 'DASHBOARD - DEMO_BUTTON',
};
