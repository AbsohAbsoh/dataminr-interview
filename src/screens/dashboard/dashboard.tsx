import {NativeStackScreenProps} from '@react-navigation/native-stack';
import React, {FC, useCallback} from 'react';
import {GestureResponderEvent, ScrollView, View} from 'react-native';
import {Button} from '../../components/Button/Button';
import {Paragraph} from '../../components/Paragraph/Paragraph';
import {useContent} from '../../config/content';
import {NavigationParams} from '../../config/navigation';
import {useStyles} from '../../config/styles';
import {dashboardStyles} from './dashboard.styles';
import {dashboardTags} from './dashboard.tags';

const DashboardScreen: FC<
  NativeStackScreenProps<NavigationParams, 'Dashboard'>
> = ({}) => {
  const styles = useStyles(dashboardStyles);
  const content = useContent('dashboard');

  const onDemoButtonPress = useCallback((_: GestureResponderEvent) => {}, []);

  return (
    <View style={styles.backgroundWrapper}>
      <ScrollView contentContainerStyle={styles.background}>
        <Paragraph
          styleOverrides={{
            paragraph: styles.demoParagraph,
          }}>
          {content.demoBody}
        </Paragraph>
        <Button
          content={content.demoButton}
          onPress={onDemoButtonPress}
          styleOverrides={{container: styles.demoButton}}
          testID={dashboardTags.demoButton}
        />
      </ScrollView>
    </View>
  );
};

export default DashboardScreen;
