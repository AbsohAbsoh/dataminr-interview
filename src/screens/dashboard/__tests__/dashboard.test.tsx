import React from 'react';
import Dashboard from '../dashboard';
import {render, RenderAPI} from '@testing-library/react-native';
import {dashboardTags} from '../dashboard.tags';

describe('Screen -- Dashboard', () => {
  let rendered: RenderAPI;
  beforeEach(() => {
    // @ts-ignore -- missing navigation dependency
    rendered = render(<Dashboard />);
  });

  it('Should render the dashboard and match the snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('OK button should be found in render', async () => {
    const button = rendered.getByTestId(dashboardTags.demoButton);
    expect(button).toBeTruthy();
  });
});
