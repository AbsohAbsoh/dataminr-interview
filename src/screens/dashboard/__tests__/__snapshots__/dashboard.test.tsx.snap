// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen -- Dashboard Should render the dashboard and match the snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffec",
      "flex": 1,
      "padding": 25,
      "width": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#333333",
          "fontSize": 18,
          "lineHeight": 30,
          "marginTop": 150,
          "textAlign": "center",
        }
      }
    >
      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English.
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff5800",
          "borderRadius": 40,
          "marginHorizontal": 0,
          "marginTop": 150,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 1,
          "width": 200,
        }
      }
      testID="DASHBOARD - DEMO_BUTTON"
    >
      <Text
        style={
          Object {
            "color": "#2a2a2a",
            "fontSize": 30,
          }
        }
      >
        OK
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
